### PR TITLE
Character gets FractionEnum

### PR DESCRIPTION
**Pull request by Marco5de**

### CODE REVIEW

**Review by ottojo — APPROVED** [review]

File: src/datatypes/character/Character.cpp [hunk 1]
```diff
@@ -134,4 +134,13 @@ namespace spy::character {
     bool Character::hasProperty(PropertyEnum property) const {
         return std::find(properties.begin(), properties.end(), property) != properties.end();
     }
+
+    void Character::setFraction(character::FactionEnum f) {
+        this->fraction = f;
```
Comment (authaldo) [inline src/datatypes/character/Character.cpp]: typo?

```suggestion
        this->faction = f;
```

File: src/datatypes/character/Character.cpp [hunk 2]
```diff
@@ -134,4 +134,13 @@ namespace spy::character {
...
+    }
+
+    character::FactionEnum Character::getFraction() const {
+        return fraction;
```
Comment (authaldo) [inline src/datatypes/character/Character.cpp]: typo?

```suggestion
        return faction;
```

File: src/datatypes/character/Character.hpp [hunk 3]
```diff
@@ -73,6 +73,10 @@ namespace spy::character {
 
             void setGadgets(const std::vector<gadget::Gadget> &gadgets);
 
+            void setFraction(character::FactionEnum fraction);
```
Comment (authaldo) [inline src/datatypes/character/Character.hpp]: typo?

```suggestion
            void setFaction(character::FactionEnum faction);
```

File: src/datatypes/character/Character.hpp [hunk 4]
```diff
@@ -73,6 +73,10 @@ namespace spy::character {
...
+
+            [[nodiscard]] character::FactionEnum getFraction() const;
```
Comment (authaldo) [inline src/datatypes/character/Character.hpp]: typo?

```suggestion
            [[nodiscard]] character::FactionEnum getFaction() const;
```

**Review by authaldo — CHANGES_REQUESTED** [review]

File: src/datatypes/character/Character.hpp [hunk 5]
```diff
@@ -94,6 +98,7 @@ namespace spy::character {
             unsigned int chips = DEFAULT_CHIPS;
             std::set<spy::character::PropertyEnum> properties;
             std::vector<spy::gadget::Gadget> gadgets;
+            character::FactionEnum fraction = FactionEnum::INVALID;
```
Comment (authaldo) [inline src/datatypes/character/Character.hpp]:
```suggestion
            character::FactionEnum faction = FactionEnum::INVALID;
```

**Review by authaldo — CHANGES_REQUESTED** [review]

**Review by authaldo — APPROVED** [review]

**Pull request merged**